stock_quant_view_reservation/views/stock_quant.xml

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2020 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record model="ir.ui.view" id="stock_quant_tree_view">
+        <field name="name">stock.quant.tree (in stock_quant_view_reservations)</field>
+        <field name="model">stock.quant</field>
+        <field name="inherit_id" ref="stock.view_stock_quant_tree_editable" />
```

Comment:
@xavier-bouquiaux Tested functionally and the button does not appear.

IMHO You should also add it in normal tree view:

https://github.com/odoo/odoo/blob/14.0/addons/stock/views/stock_quant_views.xml#L152